crates/saya/core/src/lib.rs

```diff
@@ -196,30 +198,39 @@ impl Saya {
             state_updates: state_updates_to_prove,
         };
 
-        println!("Program input: {}", new_program_input.serialize()?);
-
-        // let to_prove = ProvedStateDiff {
-        //     genesis_state_hash,
-        //     prev_state_hash: prev_block.header.header.state_root,
-        //     state_updates: state_updates_to_prove,
-        // };
+        let world_da = new_program_input.da_as_calldata(self.config.world_address);
+        let world_da_printable: Vec<String> = world_da.iter().map(|x| x.to_string()).collect();
+        trace!(target: "saya_core", "World DA {world_da_printable:?}.");
 
         trace!(target: "saya_core", "Proving block {block_number}.");
-        let proof = prover::prove(new_program_input.serialize()?, self.config.prover).await?;
+        let proof =
+            prove_stone(new_program_input.serialize(Some(self.config.world_address))?).await?;
```

Comment:
Out of the scope of this PR.

During the next iteration, we should try to use the input configuration to actually choose the prover. And the first one to be available for this would be:
1. Local stone.
2. Http stone.